torch/nested/_internal/nested_tensor.py

```diff
@@ -129,6 +130,12 @@ def __tensor_unflatten__(inner_tensors: Dict, meta):
         values = inner_tensors["_values"]
         offsets = inner_tensors["_offsets"]
 
+        should_prop = any(any(is_symbolic(s) for s in x.shape) for x in (values, offsets))
```

Comment:
Why not just check is_fake?

Reply:
So the reason is that `is_fake(FunctionalTensor())` returns False, but can hold symbolic sizes. I think its holding a fake tensor underneath? but since it's not considered a desugaring subclass, `is_fake` does not understand that.